.github/workflows/tests.yaml

```diff
@@ -0,0 +1,38 @@
+name: test
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - "**"
+    paths-ignore:
+      - "docs/**"
+
+concurrency:
+  group: build-test-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  test:
+    name: "Unit and Integration Tests"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.10'
+      - name: Install system packages
+        run: sudo apt-get install -y portaudio19-dev
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+          pip install -r requirements-full.txt
+      - name: Test with pytest
+        run: |
+          pip install pytest
+          pytest src/dailyai/tests/
```

Comment:
There's a `load_tests` function in `test_aggregators.py` that loads the `aggregators` module and then runs a `DocTestSuite` against that module. It doesn't seem to be picked up by pytest. (which is why pytest is reporting fewer tests than unittest). Not sure if there's a way to address that directly with pytest or if we should run unittest.

Reply:
For the record, the way to run it with unitest is:

```
python -m unittest -v src/dailyai/tests/test_*
```

However, `pytest` doesn't like `load_tests` and since we are only loading the doctests, we can just do:

```
pytest --doctest-modules --ignore-glob="*to_be_updated*" src/dailyai
```

Reply:
See https://github.com/pytest-dev/pytest/issues/992